apps/src/code-studio/components/progress/LessonProgress.jsx

```diff
@@ -50,13 +76,69 @@ const styles = {
  */
 class LessonProgress extends Component {
   static propTypes = {
-    // redux provided
     levels: PropTypes.arrayOf(levelType).isRequired,
     lessonExtrasUrl: PropTypes.string,
     onLessonExtras: PropTypes.bool,
-    lessonTrophyEnabled: PropTypes.bool
+    lessonTrophyEnabled: PropTypes.bool,
+    width: PropTypes.number,
+    setDesiredWidth: PropTypes.func
   };
 
+  setDesiredWidth() {
+    const fullWidth = $('.full_progress_inner').width();
+    if (this.props.setDesiredWidth) {
+      this.props.setDesiredWidth(fullWidth);
+    }
+  }
+
+  componentDidMount() {
+    this.setDesiredWidth();
+  }
+
+  componentDidUpdate() {
+    this.setDesiredWidth();
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return this.props.width !== nextProps.width;
+  }
+
+  getFullProgressOffset() {
+    // We want to set the offset so that the current level is in the middle
+    // of the available width.
+
+    if (this.refs.currentLevel && this.props.width) {
+      const fullWidth = $('.full_progress_inner').width();
+      const actualWidth = this.props.width - 10;
+      const currentLevelOffset = $(this.refs.currentLevel).position().left;
+
+      if (fullWidth > actualWidth) {
+        let desiredOffset = actualWidth / 2 - currentLevelOffset - 34 / 2;
```

Comment:
nit: why not just `- 17` here?

Reply:
Oh, 34 is the width of the highlighted bubble, and I wanted to show my working.  I'll extract this into a variable with a comment to explain.